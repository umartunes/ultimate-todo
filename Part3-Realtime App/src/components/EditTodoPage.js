import React from 'react';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Link } from 'react-router-dom';
import TodoForm from './TodoForm';
import SideNave from './SideNave';
import { startEditTodo } from '../redux/actions/actions-todos';



export class EditTodoPage extends React.Component {

    onSubmit = (todo) => {
        this.props.startEditTodo(this.props.todos.id, todo);
        this.props.history.push('/');
    };

    render() {
        return (
            <main id="to-do_main">
                <SideNave />
                <div className="row">
                    <div className="container">
                        <div className="col s12">
                            <h5 className="white-text center-align">Edit Todo</h5>
                            <Link to="/" className="backButton white-text">
                                <FontAwesomeIcon icon="long-arrow-alt-left" className=" fs-32 white-text" size="2x" />
                            </Link>
                        </div>
                    </div>
                </div>
                <TodoForm
                    data={this.props.todos}
                    onSubmit={this.onSubmit}
                />
            </main>
        );
    };
}


const mapStateToProps = (state, props) => ({
    todos: state.todos.todos.find((todo) => todo.id == props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditTodo: (id, todo) => dispatch(startEditTodo(id, todo)),

})
export default connect(mapStateToProps, mapDispatchToProps)(EditTodoPage);