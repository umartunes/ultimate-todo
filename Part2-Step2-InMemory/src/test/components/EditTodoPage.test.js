import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import EditTodoPage from '../../components/EditTodoPage';
import { Provider } from "react-redux";
import store from '../../config/store';



test('Should Render Edit Todo Page  Correctly', () => {

    const renderer = new ReactShallowRenderer();
    renderer.render(
        < Provider store={store}>
            <EditTodoPage />
        </Provider>
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
})