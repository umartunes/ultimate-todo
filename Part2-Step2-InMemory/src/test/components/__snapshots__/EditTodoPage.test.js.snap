// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Edit Todo Page  Correctly 1`] = `<Connect(EditTodoPage) />`;
